ctapipe/reco/HillasReconstructor.py

```diff
@@ -133,12 +133,16 @@ class are set to np.nan
 
         # filter warnings for missing obs time. this is needed because MC data has no obs time
         warnings.filterwarnings(action='ignore', category=MissingFrameAttributeWarning)
-        
-        # stereoscopy needs at least two telescopes
-        if len(hillas_dict) < 2:
+
+        # stereoscopy needs at least two telescopes with valid widths
+        valid_telescopes = sum([1 if not np.isnan(hillas_dict[x]['width'].value)
```

Comment:
I think you could directly build a new dict here with all the valid telescopes and only use those later.

E.g. `valid_hillas = {tel_id: hillas for tel_id, hillas in hillas_dict.items() if np.isfinite(hillas['width'].value}`

Reply:
I think this would be the cleaner approach for now but it kind of comes back to #772.
Do we still use these images by adjusting the weights or do we not use them at all?

Reply:
Exactly, thanks for digging it out.
My feeling is that 3 aligned pixels still give information about the direction and should not simply be thrown out - but I have not done a study on that ;-)
I see the problem with weights, however, and have no clear fix to handle this.

Reply:
yes, it's nice to keep track of what telescopes were dropped. Generally, i think a real pipeline would use a CutFlow to select telescopes before Hillas parameterization (e.g. with a minimum width and length or number of surviving pixels), but the code should still do reasonable things without that separate step, so testing for at least NaN is good. It should only skip telescopes if there is a real problem, and leave it up to the user to make any looser cuts for robustness.

Reply:
and by real problem I mean e.g. a singular matrix. In that case you could still reconstruct a direction if possible, and set the h_max to NaN